### ISSUE TITLE
Designing "plugin" interface

### ISSUE THREAD

**Issue by odashi** [google/latexify_py#165]
## Description

We eventually need some mechanism to apply separate rules.

## Ideas of the solution

Provide a way to implement "plugins", that provides

* Additional AST transformer
* Additional expression rules

## Additional context

- [Previous discussion](https://github.com/google/latexify_py/issues/142#issuecomment-1345553587)

**Comment by odashi** [comment]
I don't plan to fix the final interface of plugins until the v1 release and we can start with ad-hoc implementations.

For now, the most significant issue is how we define additional expression rules for `Call`. We can start with only defining the interface to provide additinal function rules.

**Comment by ZibingZhang** [comment]
I believe `array, ndarray, zeros, identity` cases in `ExprCodegen.visit_Call` should belong in `latexify.plugins.numpy.array`

**Comment by odashi** [comment]
Yes I think so. Maybe `sum` and `prod` are also the candidates (and we can include them as a default plugin)
